src/lib/storage/simdCAI_segment.cpp

```diff
@@ -0,0 +1,94 @@
+#include "simdCAI_segment.hpp"
+
+#include <algorithm>
+
+#include "resolve_type.hpp"
+#include "utils/performance_warning.hpp"
+#include "utils/size_estimation_utils.hpp"
+
+namespace opossum {
+
+template <typename T, typename U>
+SIMDCAISegment<T, U>::SIMDCAISegment(const std::shared_ptr<const pmr_vector<uint32_t>>& encoded_values,
+                                       std::optional<pmr_vector<bool>> null_values,
+                                       uint8_t codec_id,
+                                       ChunkOffset size)
+    : AbstractEncodedSegment(data_type_from_type<T>()),
+      _encoded_values{encoded_values},
+      _null_values{null_values},
```

Comment:
By value? Could it happen that you copy the null values? (FYI: there are not NULLs in TPC-H)